src/arrayMethodJoin.js

```diff
@@ -5,7 +5,24 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    if (arguments.length !== 0 && separator !== undefined) {
```

Comment:
```suggestion
    if (arguments.length !== 0 && separator !== undefined) {
```
you don't need outer if statement